modules/layers/src/layers/editable-h3-layer.ts

```diff
@@ -0,0 +1,152 @@
+            features: [],
+          },
+          selectedFeatureIndexes: this.props.selectedFeatureIndexes || [],
+
+          onEdit: (editAction) => {
+            const { editType, editContext } = editAction;
+
+            switch (editType) {
+              case 'updateTentativeFeature':
+                // tentative feature updates, updated on every pointer move
+                if (editContext.feature.geometry.type === 'Polygon') {
+                  const coords = editContext.feature.geometry.coordinates;
+                  const hexIDs = this.getDerivedHexagonIDs(coords);
+
+                  this.setState({ tentativeHexagonIDs: hexIDs });
+                }
+                break;
+              case 'addFeature':
+                // once the shape is finished drawing, calculate final hexagons
+                let finalHexagonIDs;
+                const committedHexagonIDs = new Set(this.getHexIdsFromH3Data());
+                const tentativeHexagonIDs = new Set(this.state.tentativeHexagonIDs);
+
+                switch (this.props.modeConfig.booleanOperation) {
+                  case 'union':
+                    finalHexagonIDs = [
+                      ...new Set([...committedHexagonIDs, ...tentativeHexagonIDs]),
+                    ];
+                    break;
+                  case 'intersection':
+                    finalHexagonIDs = [...committedHexagonIDs].filter((hexID: string) =>
+                      tentativeHexagonIDs.has(hexID)
+                    );
+                    break;
+                  case 'difference':
+                    finalHexagonIDs = [...committedHexagonIDs].filter(
+                      (hexID: string) => !tentativeHexagonIDs.has(hexID)
+                    );
+                    break;
+                  default:
+                    break;
+                }
+
+                this.setState({
+                  tentativeHexagonIDs: [],
+                });
+
+                this.props.onEdit({ updatedData: finalHexagonIDs });
+                break;
+              default:
+                break;
+            }
+          },
+        })
+      ),
+
+      new H3ClusterLayer(
+        this.getSubLayerProps({
+          id: 'tentative-hexagons',
+          data: [
+            {
+              count: this.state.tentativeHexagonIDs.length,
+              hexIds: this.state.tentativeHexagonIDs,
+            },
+          ],
+          getHexagons: (d) => d.hexIds,
+        })
+      ),
+      new H3ClusterLayer(
+        this.getSubLayerProps({
+          id: 'hexagons',
+          data: this.props.h3Data,
+          getHexagons: this.props.getHexagons,
+        })
+      ),
+    ];
+    return layers;
+  }
+
+  // because h3Data is an array of hexagon data, we take the cumulative hexIds from all entries
+  getHexIdsFromH3Data() {
+    let cumulativeHexIds = [];
+    this.props.h3Data.forEach(
+      (h3Data) => (cumulativeHexIds = cumulativeHexIds.concat(h3Data.hexIds))
```

Comment:
Seems you should use `this.props.getHexagons` instead of hardcoding here.

Reply:
That's fair. This behavior should be left up to the caller.